src/js/player.js

```diff
@@ -823,6 +825,112 @@ class Player extends Component {
   }
 
   /**
+   * Modify the video/audio element so that we can detect when
+   * the source is changed. Fires `sourceset` just after the source has changed
+   *
+   * @fires Player#sourceset
+   */
+  watchForSourceSet_() {
```

Comment:
This change is basically HTML5 only and should probably be moved into the html5 tech. However, this leaves us with the question of what to do for other techs like Flash?
Also, what to do if we're on a browser like iOS 7? Should we fallback and trigger `sourceset` at the earliest opportunity that we know the source has changed otherwise? For example on a `loadstart` if we haven't already triggered a `sourceset`? It's possible that this also answers the Flash-tech issue.

Reply:
Yeah this does seem like it should be in HTML5. I would think that we should try to support sourceset as well as we can in older browsers and other techs, but I think we may want to do that in another PR as this is already quite big.

Reply:
That sounds reasonable. As long as we have an idea of a path forward.